Wrappers/Python/test/test_functions.py

```diff
@@ -918,171 +922,209 @@ def test_Lipschitz4(self):
-            
-            t2 = timer()
-            res2 = g_CCPI_reg_toolkit.proximal(noisy_data, 1.)
-            t3 = timer()
-            print(t3-t1)
-            
-            
-            np.testing.assert_array_almost_equal(res1.as_array(), res2.as_array(), decimal = 4)
-            
-            ###################################################################
-            ###################################################################
-            ###################################################################
-            ###################################################################    
-            
-            print("Compare CIL_FGP_TV vs CCPiReg_FGP_TV with iterations.")
-            iters = 408
-            # CIL_FGP_TV no tolerance
-            g_CIL = TotalVariation(alpha, iters, tolerance=1e-9, lower = 0.)
-            t0 = timer()
-            res1 = g_CIL.proximal(noisy_data, 1.)
-            t1 = timer()
-            print(t1-t0)
+    def test_rmul(self):
+        alpha = 0.15
+        tv = alpha * TotalVariation()
+        assert isinstance(tv, TotalVariation)
+    
+    def test_regularisation_parameter3(self):
+        tv = TotalVariation()
+        try:
+            tv.regularisation_parameter = 'string'
+            assert False
+        except TypeError as te:
+            print (te)
+            assert True
+    def test_rmul2(self):
+        alpha = 'string'
+        try:
+            tv = alpha * TotalVariation()
+            assert False
+        except TypeError as te:
+            print (te)
+            assert True
+    
+
+    @unittest.skipUnless(has_reg_toolkit, "Regularisation Toolkit not present")
+    def test_compare_regularisation_toolkit(self):
+    
+        print("Compare CIL_FGP_TV vs CCPiReg_FGP_TV no tolerance (2D)")
+
+        data = dataexample.SHAPES.get()
+        ig = data.geometry
+        ag = ig
+
+        # Create noisy data. 
+        n1 = np.random.normal(0, 0.1, size = ig.shape)
+        noisy_data = ig.allocate()
+        noisy_data.fill(n1+data.as_array())
+        
+        alpha = 0.1
+        iters = 1000
             
-            # CCPi Regularisation toolkit high tolerance
-            r_alpha = alpha
-            r_iterations = iters
-            r_tolerance = 1e-9
-            r_iso = 0
-            r_nonneg = 1
-            r_printing = 0
-            g_CCPI_reg_toolkit = CCPiReg_FGP_TV(r_alpha, r_iterations, r_tolerance, r_iso, r_nonneg, r_printing, 'cpu')
-
-            t2 = timer()
-            res2 = g_CCPI_reg_toolkit.proximal(noisy_data, 1.)
-            t3 = timer()
-            print(t3-t2)
+        # CIL_FGP_TV no tolerance
+        g_CIL = alpha * TotalVariation(iters, tolerance=None, lower = 0, info = True)
+        t0 = timer()
+        res1 = g_CIL.proximal(noisy_data, 1.)
+        t1 = timer()
+        print(t1-t0)
+        
+        # CCPi Regularisation toolkit high tolerance
+        r_alpha = alpha
+        r_iterations = iters
```

Comment:
Do we have to do 1000 iterations? if we are comparing the 2 implementations couldn't we do a couple and expect the same answers between them?

Reply:
Same in other tests.

Reply:
Yes, I thought so too and I had the test with much fewer iterations, but the unittest failed. 

So what shall we do? I suppose this test checks 2 implementations that should give the same asymptotic result?